qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferModel.cpp

```diff
@@ -0,0 +1,414 @@
+// Mantid Repository : https://github.ervicecom/mantidproject/mantid
+//
+// Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ISISEnergyTransferModel.h"
+#include "ISISEnergyTransferModelUtils.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AlgorithmRuntimeProps.h"
+#include "MantidAPI/MatrixWorkspace.h"
+
+#include <filesystem>
+
+using namespace Mantid::API;
+
+namespace MantidQt::CustomInterfaces {
+IETModel::IETModel() {}
+
+std::vector<std::string> IETModel::validateRunData(IETRunData const &runData, std::size_t const &defaultSpectraMin,
+                                                   std::size_t const &defaultSpectraMax) {
+  std::vector<std::string> errors;
+  IETDataValidator validator;
+
+  auto inputFiles = runData.getInputData().getInputFiles();
+  bool isRunFileValid = !inputFiles.empty();
+  std::string firstFileName = inputFiles.substr(0, inputFiles.find(','));
+
+  std::string detectorError =
+      validator.validateDetectorGrouping(runData.getGroupingData(), defaultSpectraMin, defaultSpectraMax);
+  if (!detectorError.empty()) {
+    errors.push_back(detectorError);
+  }
+
+  std::string analysisError = validator.validateAnalysisData(runData.getAnalysisData());
+  if (!analysisError.empty()) {
+    errors.push_back(analysisError);
+  }
+
+  std::string conversionError = validator.validateConversionData(runData.getConversionData());
+  if (!conversionError.empty()) {
+    errors.push_back(conversionError);
+  }
+
+  std::vector<std::string> backgroundErrors = validator.validateBackgroundData(
+      runData.getBackgroundData(), runData.getConversionData(), firstFileName, isRunFileValid);
+  errors.insert(errors.end(), backgroundErrors.begin(), backgroundErrors.end());
+
+  errors.erase(std::remove(errors.begin(), errors.end(), ""), errors.end());
+
+  return errors;
+}
+
+void IETModel::setInstrumentProperties(IAlgorithm_sptr const &reductionAlg, InstrumentData const &instData) {
+  std::string instrument = instData.getInstrument();
+  std::string analyser = instData.getAnalyser();
+  std::string reflection = instData.getReflection();
+
+  reductionAlg->setProperty("Instrument", instrument);
```

Comment:
could do these in one line
```suggestion
  reductionAlg->setProperty("Instrument",  instData.getInstrument());
```